sim/task.h

```diff
@@ -65,11 +70,11 @@ typedef void (*TaskFunction_t)(void *instance);
  * \ingroup Tasks
  */
 //typedef void * TaskHandle_t;
-struct TaskHandle_t {
-  void *thread_handle = nullptr;
+typedef struct TaskHandle_t {
+  void *thread_handle;
   TaskFunction_t task_fn;
-  void *instance = nullptr;
-};
+  void *instance;
+}TaskHandle_t;
```

Comment:
nitpick: space before name

```suggestion
} TaskHandle_t;
```